joblib/_parallel_backends.py

```diff
@@ -577,8 +597,17 @@ def abort_everything(self, ensure_ready=True):
         """Shutdown the workers and restart a new one with the same parameters
         """
         self._workers.shutdown(kill_workers=True)
-        delete_folder(self._workers._temp_folder)
+        # The folder can be safely deleted without risk of PermissionError in
+        # Windows as all workers were killed.
+        delete_folder(self._workers._temp_folder, allow_non_empty=True)
         self._workers = None
+
+        # All temporary files have been deleted -- unregister the temporary
+        # files from the resource_tracker refcount tables.
+        for filename in JOBLIB_MMAPS:
+            resource_tracker.unregister(filename, "file")
+        JOBLIB_MMAPS.clear()
```

Comment:
Similar remark here: we list the files in `self._workers._temp_folder` prior to deleting them and then unregister them.